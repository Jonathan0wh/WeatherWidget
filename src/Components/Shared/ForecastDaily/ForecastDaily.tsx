import React from 'react';
import { StyleSheet, View, Text, Image } from 'react-native';
import { useSelector } from 'react-redux';

import { COLORS } from 'styles/colors';
import { A11Y_LABELS } from 'constants/a11y';
import { STRINGS } from 'constants/strings';
import { RootState } from 'store/reducer';
import { getFahrenheitFromCelsius, getWeatherIconUri } from 'utils';
import { DailyForecastState } from 'Views/Forecast/types';

interface ForecastDailyProps {
  index: number;
  dailyForecast: DailyForecastState;
}

const ForecastDaily = ({ index, dailyForecast }: ForecastDailyProps) => {
  const current = useSelector((state: RootState) => state.current);
  const celsius = useSelector((state: RootState) => state.unit.celsius);

  const dayIndex = current ? new Date().getDay() : 0;

  return (
    <View
      style={styles.viewContainer}
      accessible
      accessibilityLabel={A11Y_LABELS.forecast}
      accessibilityRole="text"
    >
      <Text style={styles.dateText}>
        {index
          ? current &&
            STRINGS.days[
              dayIndex + index >= 7 ? dayIndex + index - 7 : dayIndex + index
            ]
          : 'Today'}
      </Text>
      <Image
        source={{
          uri: getWeatherIconUri(dailyForecast.weather.icon)
        }}
        style={styles.weatherIcon}
        accessible
        accessibilityLabel={A11Y_LABELS.forecastDailyWeather}
      />
      <View style={styles.rowContainer}>
        <Text style={styles.highTemperatureText}>
          {celsius
            ? dailyForecast.max_temp.toFixed(0)
            : getFahrenheitFromCelsius(dailyForecast.max_temp)}
          °
        </Text>
        <Text style={styles.infoText}>
          {' '}
          {celsius
            ? dailyForecast.min_temp.toFixed(0)
            : getFahrenheitFromCelsius(dailyForecast.min_temp)}
          °
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  viewContainer: {
    padding: 10
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'center'
  },
  dateText: {
    fontWeight: 'bold'
  },
  infoText: {
    color: COLORS.infoTextGrey
  },
  highTemperatureText: {
    color: COLORS.titleTextGrey
  },
  weatherIcon: {
    width: 60,
    height: 60
  }
});

export default ForecastDaily;
