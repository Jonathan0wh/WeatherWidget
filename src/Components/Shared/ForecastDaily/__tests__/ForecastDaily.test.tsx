import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ForecastDaily from '../ForecastDaily';

const mockStore = configureMockStore([thunk]);
const mockResponseForecast = require('./response.json');
const mockResponseDailyForecast = mockResponseForecast.data[0];

describe('<ForecastDaily />', () => {
  it('should render by default', () => {
    const store = mockStore({
      location: null,
      current: null,
      unit: {
        celsius: true
      }
    });
    const wrapper = shallow(
      <Provider store={store}>
        <ForecastDaily index={0} dailyForecast={mockResponseDailyForecast} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with Farenheight', () => {
    const store = mockStore({
      location: {},
      current: {},
      unit: {
        celsius: false
      }
    });
    const wrapper = shallow(
      <Provider store={store}>
        <ForecastDaily index={0} dailyForecast={mockResponseDailyForecast} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
