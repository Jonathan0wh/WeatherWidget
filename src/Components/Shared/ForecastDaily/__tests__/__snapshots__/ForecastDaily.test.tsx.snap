// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastDaily /> should render by default 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ForecastDaily
    dailyForecast={
      Object {
        "app_max_temp": 22.7,
        "app_min_temp": 20.7,
        "clouds": 100,
        "clouds_hi": 100,
        "clouds_low": 99,
        "clouds_mid": 100,
        "datetime": "2020-02-09",
        "dewpt": 20.1,
        "high_temp": 22.7,
        "low_temp": 21.1,
        "max_dhi": null,
        "max_temp": 22.2,
        "min_temp": 20.1,
        "moon_phase": 0.994843,
        "moonrise_ts": 1581239723,
        "moonset_ts": 1581192062,
        "ozone": 265.339,
        "pop": 95,
        "precip": 112.562,
        "pres": 1001.25,
        "rh": 93,
        "slp": 1008.84,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1581189984,
        "sunset_ts": 1581238824,
        "temp": 21.2,
        "ts": 1581166860,
        "uv": 1.34279,
        "valid_date": "2020-02-09",
        "vis": 3.14621,
        "weather": Object {
          "code": 202,
          "description": "Thunderstorm with heavy rain",
          "icon": "t03d",
        },
        "wind_cdir": "E",
        "wind_cdir_full": "east",
        "wind_dir": 83,
        "wind_gust_spd": 21.0305,
        "wind_spd": 12.608,
      }
    }
    index={0}
  />
</ContextProvider>
`;

exports[`<ForecastDaily /> should render with Farenheight 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ForecastDaily
    dailyForecast={
      Object {
        "app_max_temp": 22.7,
        "app_min_temp": 20.7,
        "clouds": 100,
        "clouds_hi": 100,
        "clouds_low": 99,
        "clouds_mid": 100,
        "datetime": "2020-02-09",
        "dewpt": 20.1,
        "high_temp": 22.7,
        "low_temp": 21.1,
        "max_dhi": null,
        "max_temp": 22.2,
        "min_temp": 20.1,
        "moon_phase": 0.994843,
        "moonrise_ts": 1581239723,
        "moonset_ts": 1581192062,
        "ozone": 265.339,
        "pop": 95,
        "precip": 112.562,
        "pres": 1001.25,
        "rh": 93,
        "slp": 1008.84,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1581189984,
        "sunset_ts": 1581238824,
        "temp": 21.2,
        "ts": 1581166860,
        "uv": 1.34279,
        "valid_date": "2020-02-09",
        "vis": 3.14621,
        "weather": Object {
          "code": 202,
          "description": "Thunderstorm with heavy rain",
          "icon": "t03d",
        },
        "wind_cdir": "E",
        "wind_cdir_full": "east",
        "wind_dir": 83,
        "wind_gust_spd": 21.0305,
        "wind_spd": 12.608,
      }
    }
    index={0}
  />
</ContextProvider>
`;
